os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c

```diff
@@ -408,7 +408,7 @@ static int imxrt_gpt_ioctl(struct timer_lowerhalf_s *lower, int cmd,
 			ret = (int)imxrt_gpt_getclockdivider(priv->gpt->base);
 		}
 		break;
-	case TCIOC_SETFREERUN:
+	case TCIOC_SETMODE:
```

Comment:
Most of timer provides at least 3 mode (one-shot, repeat, free-run, some h/w supports free-run-interrupt)
so boolean is not good option for arg.
You should define some hex value of these modes. (consider about free-run & free-run-interrupt, hex will be better for bit comparing)